node/src/blockchain/errors/validation_status.rs

```diff
@@ -38,7 +64,7 @@ impl PreviousAttempts {
     }
 }
 
-#[derive(Debug, Clone, PartialEq, Eq, Serialize, Deserialize)]
+#[derive(Debug, Clone, PartialEq, Eq, Hash, Serialize, Deserialize, PartialOrd, Ord)]
```

Comment:
If I examine that acted on your own as to the replacement of vecs to BTreeSets, since nobody of us others had never complained, you caused a lot of problems thanks to this experiment...